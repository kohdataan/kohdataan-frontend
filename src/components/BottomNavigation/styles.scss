@import '../../styles/colors.scss';

.bottom-navigation {
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  justify-content: center;
  height: 64px;
  background-color: $black;
  border-top: 1px solid $orange;
  box-shadow: 0 5px 0px 0 rgba(0, 0, 0, 0.16), 0 1px 5px 0 rgba(0, 0, 0, 0.26);
  z-index: 3000;
  width: 100%;
  max-width: 960px;
}

.bottom-navigation-container {
  width: 100%;
  position: fixed;
  height: 64px;
  bottom: 0;
  left: 0;
  display: flex;
  justify-content: center;
}
