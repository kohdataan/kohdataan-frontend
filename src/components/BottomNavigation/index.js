import React, { memo } from 'react'
import PropTypes from 'prop-types'
import './styles.scss'

const BottomNavigation = props => {
  const { children, className, ...other } = props
  return (
    <div className="bottom-navigation-container">
      <nav className={className} {...other}>
        {children}
      </nav>
    </div>
  )
}

BottomNavigation.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
}

BottomNavigation.defaultProps = {
  className: 'bottom-navigation',
}

export default memo(BottomNavigation)
