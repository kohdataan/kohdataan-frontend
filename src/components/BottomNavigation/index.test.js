import React from 'react'
import { create } from 'react-test-renderer'
import BottomNavigation from './index'

describe('Bottom navigation', () => {
  test('it matches the snapshot', () => {
    const component = create(
      <BottomNavigation>
        <p>Navigation</p>
      </BottomNavigation>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
