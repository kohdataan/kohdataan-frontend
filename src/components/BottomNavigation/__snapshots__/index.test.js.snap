// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bottom navigation it matches the snapshot 1`] = `
<div
  className="bottom-navigation-container"
>
  <nav
    className="bottom-navigation"
  >
    <p>
      Navigation
    </p>
  </nav>
</div>
`;
